### PR TITLE
Kylan/show account balance tx list

### PR DESCRIPTION
**Pull request by kylanhurt**
The purpose of this task is to add the 'Show Balance' functionality and persistence to the Transaction List scene. This supplements the earlier task of adding persistence to the Wallet List scene.

Asana Task: https://app.asana.com/0/361770107085503/498144750192820

### CODE REVIEW

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 1]
```diff
@@ -47,25 +48,23 @@ export type StateProps = {
...
 type State = {
   focused: boolean,
-  balanceBoxVisible: boolean,
   op: any,
   animation: any,
-  balanceBoxOpacity: any,
-  balanceBoxHeight: any,
   width: ?number,
-  reset: boolean,
-  showBalance: boolean
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: why did you get rid of these?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 2]
```diff
@@ -75,27 +74,15 @@ const SEND_TEXT = s.strings.fragment_send_subtitle
...
-
   constructor (props: Props) {
     super(props)
+    this.state = {
+      reset: true,
+      focused: false,
+      animation: new Animated.Value(0),
+      op: new Animated.Value(0),
+      width: undefined
+    }
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: why did the state of this component change?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 3]
```diff
@@ -115,77 +102,6 @@ export class TransactionList extends Component<Props, State> {
...
-  //     ]).start(() => this.setState({ balanceBoxVisible: false }))
-  //   }
-  // }
-  //
-  // _onCancel = () => this.setState({ width: 0 })
-
-  toggleShowBalance = () => {
-    this.setState({ showBalance: !this.state.showBalance })
-  }
-
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: where did all of this go?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 4]
```diff
@@ -208,7 +119,7 @@ export class TransactionList extends Component<Props, State> {
             <View style={styles.container}>
               <View style={styles.transactionsWrap}>
                 <FlatList
-                  ListHeaderComponent={this.currentRenderBalanceBox}
+                  ListHeaderComponent={this.currentRenderBalanceBox()}
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: what is currentRenderBalanceBox and why did it change to a function?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 5]
```diff
@@ -265,77 +179,71 @@ export class TransactionList extends Component<Props, State> {
...
         </Gradient>
-      </Animated.View>
+      </TouchableOpacity>
     )
   }
 
-  renderBalanceBoxTrue = this.renderBalanceBox(true)
-  renderBalanceBoxFalse = this.renderBalanceBox(false)
-  currentRenderBalanceBox = this.renderBalanceBoxTrue
-
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: what were these things?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 6]
```diff
@@ -265,77 +179,71 @@ export class TransactionList extends Component<Props, State> {
       fiatBalanceString = receivedFiatSymbol + ' ' + intl.formatNumber(balanceInFiat || 0, { toFixed: 2 }) + ' ' + fiatCurrencyCode
     }
     return (
-      <Animated.View style={[{ height: this.state.balanceBoxHeight }]}>
+      <TouchableOpacity onPress={this.props.toggleAccountBalanceVisibility} style={styles.touchablBalanceBox} activeOpacity={BALANCE_BOX_OPACITY}>
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: why did you get rid of the animated view?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 7]
```diff
@@ -265,77 +179,71 @@ export class TransactionList extends Component<Props, State> {
...
+                    {displayDenomination.symbol ? (
+                      <T numberOfLines={1} style={[styles.currentBalanceBoxBits, styles.symbol]}>
+                        {displayDenomination.symbol + ' '}
+                        <T numberOfLines={1}>{cryptoAmountString}</T>
+                      </T>
+                    ) : (
+                      <T numberOfLines={1} style={styles.currentBalanceBoxBits}>
+                        {cryptoAmountString}
+                      </T>
+                    )}
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: what does this change do?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 8]
```diff
@@ -265,77 +179,71 @@ export class TransactionList extends Component<Props, State> {
...
-                          </T>
-                        )}
-                      </View>
-                    </View>
-                    <View style={[styles.currentBalanceBoxDollarsWrap]}>
-                      <T numberOfLines={1} style={[styles.currentBalanceBoxDollars]}>
-                        {fiatBalanceString}
+                    {!displayDenomination.symbol && (
+                      <T numberOfLines={1} style={styles.currentBalanceBoxBits}>
+                        {' ' + selectedCurrencyCode}
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: what does this change do?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 9]
```diff
@@ -265,77 +179,71 @@ export class TransactionList extends Component<Props, State> {
...
-                    <T style={[styles.balanceHiddenText]}>{SHOW_BALANCE_TEXT}</T>
-                  </View>
-                )}
-              </TouchableOpacity>
-              <View style={[styles.requestSendRow]}>
-                <TouchableHighlight style={[styles.requestBox, styles.button]} underlayColor={styleRaw.underlay.color} onPress={Actions.request}>
-                  <View style={[styles.requestWrap]}>
-                    <Image style={{ width: 25, height: 25 }} source={requestImage} />
-                    <T style={[styles.request]}>{REQUEST_TEXT}</T>
+                    )}
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: what does this change do?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 10]
```diff
@@ -265,77 +179,71 @@ export class TransactionList extends Component<Props, State> {
...
-                    <Image style={{ width: 25, height: 25 }} source={sendImage} />
-                    <T style={styles.send}>{SEND_TEXT}</T>
-                  </View>
-                </TouchableHighlight>
+                </View>
+                <View style={[styles.currentBalanceBoxDollarsWrap]}>
+                  <T numberOfLines={1} style={[styles.currentBalanceBoxDollars]}>
+                    {fiatBalanceString}
+                  </T>
+                </View>
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: what does this change do?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 11]
```diff
@@ -265,77 +179,71 @@ export class TransactionList extends Component<Props, State> {
...
+              </TouchableHighlight>
+              <TouchableHighlight style={[styles.sendBox, styles.button]} underlayColor={styleRaw.underlay.color} onPress={Actions.scan}>
+                <View style={[styles.sendWrap]}>
+                  <Image style={{ width: 25, height: 25 }} source={sendImage} />
+                  <T style={styles.send}>{SEND_TEXT}</T>
+                </View>
+              </TouchableHighlight>
+            </View>
+          </View>
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: what does this change do?

File: src/modules/UI/scenes/TransactionList/TransactionListConnector.js [hunk 12]
```diff
@@ -56,6 +57,8 @@ const mapStateToProps = (state: State) => {
   const currencyInfo: EdgeCurrencyInfo = coreWallet.currencyInfo
   // set default requiredConfirmations to 1, so once the tx is in a block consider fully confirmed
   const requiredConfirmations = currencyInfo.requiredConfirmations ? currencyInfo.requiredConfirmations : 1
+  // eventually we will want the txList balance box visibility to be set to wallet-level setting (not account)
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionListConnector.js]: comments add clarification to existing code
dont use them as a todo list

File: src/modules/UI/scenes/TransactionList/style.js [hunk 13]
```diff
@@ -65,15 +65,33 @@ export const styles = {
   },
 
   // end of searchbar stuff
-
+  touchablBalanceBox: {
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/style.js]: please spell check your code before submitting PR's

File: src/modules/UI/scenes/TransactionList/style.js [hunk 14]
```diff
@@ -65,15 +65,33 @@ export const styles = {
...
   currentBalanceBox: {
     flex: 1,
     justifyContent: 'center'
   },
+  totalBalanceWrap: {
+    flex: 3,
+    alignItems: 'center',
+    backgroundColor: THEME.COLORS.TRANSPARENT
+  },
+  hiddenBalanceBoxDollarsWrap: {
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/style.js]: what does this do?

File: src/modules/UI/scenes/TransactionList/style.js [hunk 15]
```diff
@@ -65,15 +65,33 @@ export const styles = {
...
   // end of searchbar stuff
-
+  touchablBalanceBox: {
+    height: 200
+  },
   currentBalanceBox: {
     flex: 1,
     justifyContent: 'center'
   },
+  totalBalanceWrap: {
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/style.js]: what does this do?

File: src/modules/UI/scenes/TransactionList/style.js [hunk 16]
```diff
@@ -65,15 +65,33 @@ export const styles = {
...
+    alignItems: 'center',
+    backgroundColor: THEME.COLORS.TRANSPARENT
+  },
+  hiddenBalanceBoxDollarsWrap: {
+    flex: 3,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: THEME.COLORS.TRANSPARENT
+  },
+  currentBalanceBoxHiddenText: {
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/style.js]: what does this do?

File: src/modules/UI/scenes/TransactionList/style.js [hunk 17]
```diff
@@ -102,9 +121,9 @@ export const styles = {
     justifyContent: 'flex-start',
     backgroundColor: THEME.COLORS.TRANSPARENT
   },
-  currentBalanceBoxBitssWrap: {
+  currentBalanceBoxBitsWrap: {
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/style.js]: what is this?

File: src/modules/UI/scenes/WalletList/style.js [hunk 18]
```diff
@@ -49,7 +49,7 @@ export const styles = {
     fontSize: 44
   },
   currentBalanceBoxBits: {
-    color: THEME.COLORS.WHITE,
+    color: THEME.COLORS.TRANSPARENT,
```
Comment (wolverineks) [inline src/modules/UI/scenes/WalletList/style.js]: why did you change anything on the wallet list?

File: src/modules/UI/scenes/TransactionList/TransactionListConnector.js [hunk 19]
```diff
@@ -56,6 +57,8 @@ const mapStateToProps = (state: State) => {
...
+  const isAccountBalanceVisible = state.ui.settings.isAccountBalanceVisible
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionListConnector.js]: why are you using an account setting to control non account stuff?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 1 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: These state variables are related to defunct Search Bar that we decided not to implement (yet) approximately 6 months ago. They are legacy code that are inert.

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 3 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: All code related the defunct search feature has been removed. Additionally, `toggleShowBalance` functionality has been removed from component state to Redux (since it is a synced setting now).

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 4 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: Because, in my opinion, a function returns a component (JSX) whereas the property name `ListHeadComponent` grammatically is asking for a component. It may work either way but grammatically I think the change adds clarity.

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 6 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: Because the balance box used to need a fade (opacity change) for the search feature (that was never implemented), and since the area is now "pressable" that means we can use the `TouchableOpacity`'s native `activeOpacity` property to create the desired effect with improved performance.

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 7 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: This is the implementation of the **actual** "Show Balance" version of the box, which was **not** there before. It was present on the WalletList but not this TransactionList

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 8 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: I honestly don't even see an actual change. Maybe indentation?

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 9 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: Change in the structure of the JSX of the balance box to better create uniformity between the "Show Balance" version and the actual Balance version.

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 10 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: Not much, if anything, Git is having issues aligning the equivalent pieces of code.

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 11 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: Not much, if anything, Git is having issues aligning the equivalent pieces of code. (Look in the left column a few dozen rows up)

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js (hunk 5 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: In my opinion the previous implementation of the 'Show Balance' component (non-Redux) was "wonky" and this has been replaced with an implementation that is more straight-forward.

File: src/modules/UI/scenes/TransactionList/TransactionListConnector.js (hunk 12 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionListConnector.js]: Okay. I have added an Asana task:
https://app.asana.com/0/361770107085503/734196653255276

File: src/modules/UI/scenes/TransactionList/TransactionListConnector.js (hunk 19 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/TransactionListConnector.js]: Because for now it **is** account level. @paullinator has specifically said that, since resources and time are limited, that we will go with Account-level specificity for the time being. I have already created a task in Asana for moving the TxList "Show Balance" feature to Wallet-level.

One step at a time.

File: src/modules/UI/scenes/TransactionList/style.js (hunk 13 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/style.js]: Good catch, although I'm not sure what the best way to implement spell-check is considering the crazy nature of variable / property names.

Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/style.js]: PS the spelling fix has been added / committed.

File: src/modules/UI/scenes/TransactionList/style.js (hunk 15 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/style.js]: It is a wrap around certain elements in the TxList Balance Box.

File: src/modules/UI/scenes/TransactionList/style.js (hunk 16 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/style.js]: Since I changed some of the elements in the 'Show Balance' portion of the BalanceBox that means there had to be styling changes.

File: src/modules/UI/scenes/TransactionList/style.js (hunk 15 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/style.js]: It styles a portion of the BalanceBox. I modified the structure of the JSX and that means adjustments to the styling

File: src/modules/UI/scenes/TransactionList/style.js (hunk 17 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/style.js]: It's an adjustment to the styling and also a spelling fix

File: src/modules/UI/scenes/WalletList/style.js (hunk 18 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/WalletList/style.js]: This looks like a mistake. Thank you for the heads up.

File: src/modules/UI/scenes/TransactionList/style.js (hunk 14 above)
Comment (kylanhurt) [reply src/modules/UI/scenes/TransactionList/style.js]: It is a wrap around certain elements in the TxList Balance Box.

File: src/modules/UI/scenes/TransactionList/TransactionList.ui.js [hunk 20]
```diff
@@ -265,77 +179,71 @@ export class TransactionList extends Component<Props, State> {
...
-                            {displayDenomination.symbol + ' '}
-                            <T numberOfLines={1}>{cryptoAmountString}</T>
-                          </T>
-                        ) : (
-                          <T numberOfLines={1} style={styles.currentBalanceBoxBits}>
-                            {cryptoAmountString}
-                          </T>
-                        )}
+          <View style={styles.balanceBoxContents}>
+            {!isAccountBalanceVisible ? (
```
Comment (wolverineks) [inline src/modules/UI/scenes/TransactionList/TransactionList.ui.js]: What does this boolean do?

**Comment by kylanhurt** [comment]
@paullinator @swansontec This PR is also ready to be merged into `develop`
